modules/openapi-generator/src/test/java/org/openapitools/codegen/options/TypeScriptNodeClientOptionsProvider.java

```diff
@@ -71,6 +71,7 @@ public Map<String, String> createOptions() {
                 .put(CodegenConstants.LEGACY_DISCRIMINATOR_BEHAVIOR, "true")
                 .put(CodegenConstants.DISALLOW_ADDITIONAL_PROPERTIES_IF_NOT_PRESENT, "true")
                 .put(CodegenConstants.ENUM_UNKNOWN_DEFAULT_CASE, ENUM_UNKNOWN_DEFAULT_CASE_VALUE)
+                .put(CodegenConstants.LICENSE_NAME, AbstractTypeScriptClientCodegen.LICENSE_NAME_DEFAULT_VALUE)
```

Comment:
Just a thought: would it make at all sense to introduce an abstract options provider that has this for all current and also future typescript generators?

Reply:
It could make sense given that the Typescript generators seem to have a lot of options in common. I don't see any other abstract OptionsProviders though, so it is not a common practice in this repo. I could have a crack at it if I get the time to see what it looks like.

Reply:
I implemented a shared interface which all the Typescript OptionsProviders extend, let me know if you want me to keep it in or remove it. I didn't bother to add OptionsProviders for the typescript generators which were missing ones, there might be that some of the other generators shouldn't have the parameters in the shared interface, I haven't checked that.

Reply:
I think the change is ace. @macjohnny WDYT?